js/slideshow.js

```diff
@@ -187,6 +194,79 @@
 		},
 
 		/**
+		 * Calculates the luminance of an image
+		 *
+		 * @param {*} image
+		 * @param {string} mimeType
+		 *
+		 * @returns {string}
+		 * @private
+		 */
+		_calculateBackgroundColour: function (image, mimeType) {
+			var backgroundColour = this.darkBackgroundColour;
+			if (!this._isTransparent(image.mimeType)) {
+				return backgroundColour;
+			}
+
+			// The name has to be 'canvas'
+			var lumiCanvas = document.createElement('canvas');
+			lumiCanvas.width = 200;
```

Comment:
Instead of forcing a width of 200px and adapting the height of the canvas, one could directly scale the image to get the right number of samples on drawing:
```js
var imgArea = image.width * image.height;
var canArea = numberOfSamples;
var factor = Math.sqrt(canArea / imgArea);

var scaledWidth = factor * image.width;
var scaledHeight = factor * image.height;

lumiCanvas.width = scaledWidth;
lumiCanvas.height = scaledHeight;
var lumiCtx = lumiCanvas.getContext('2d');
lumiCtx.drawImage(image, 0, 0, scaledWidth, scaledHeight);
var imgData = lumiCtx.getImageData(0, 0, scaledWidth, scaledHeight);
var pix = imgData.data;
// pix.length will be approximately 4*numberOfSamples (for RGBA)
```